docs/source/dataset.rst

```diff
@@ -0,0 +1,234 @@
+.. _dataset:
+
+Dataset
++++++++
+
+The dataset configuration can be found :ref:`here <configurationdefaultparam>`, this file contain 
+specific information about input data parameters for the execution of your command. The documentation
+on the parameters use is explain in the :ref:`yaml <yamlparameters>` section.
+
+The sampling and inference steps requires a csv referencing input data. An example of input csv for
+massachusetts buildings dataset can be found in 
+`tests <https://github.com/NRCan/geo-deep-learning/blob/develop/tests/tiling/tiling_segmentation_binary_ci.csv>`_. 
+Each row of this csv is considered, in geo-deep-learning terms, to be an 
+`AOI <https://torchgeo.readthedocs.io/en/latest/user/glossary.html#term-area-of-interest-AOI>`_.
+
+.. list-table:: **Example CSV**
+   :header-rows: 1
+
+   * - raster path 
+     - vector ground truth path
+     - dataset split
+     - aoi id (optional)
+   * - my_dir/my_geoimagery1.tif
+     - my_dir/my_geogt1.gpkg
+     - trn
+     - Ontario-1
+   * - my_dir/my_geoimagery2.tif
+     - my_dir/my_geogt2.gpkg
+     - tst
+     - NewBrunswick-23
+   * - ...
+     - ...
+     - ...
+     - ...
+
+.. note::
+    
+     If left blank, the aoi id will be derived from the raster path and bands requested.
+
+The path to a custom csv must be entered in the 
+`dataset configuration <https://github.com/NRCan/geo-deep-learning/blob/develop/config/dataset/test_ci_segmentation_binary.yaml>`_.
+See the :ref:`yaml <yamlparameters>` section for more information.
+
+Dataset splits
+--------------
+
+Split in csv should be either "trn", "tst" or "inference". The validation split is automatically 
+created during training and taken from the "trn" set. Its proportion is set by the 

```

Comment:
@mpelchat04 I think the previous description was correct. The validation split is created during tiling from input images/labels tagged "trn":
https://github.com/NRCan/geo-deep-learning/blob/develop/tiling_segmentation.py#L523
The tiling script outputs lists of patches for "trn", "val" and "tst" and these lists are used as is during training.

Reply:
Previous description was :
```
Split in csv should be either "trn", "tst" or "inference". The validation split is automatically 
created during tiling.
```
I only wanted to specify that the val split is taken from the 'trn' set.